src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -282,6 +282,27 @@ protected int getCacheLimit() {
 
     }
 
+    @CoreMethod(names = "instantiable?", isModuleFunction = true, required = 1)
+    public abstract static class InstantiableNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public boolean isInstantiable(
+                TruffleObject receiver,
+                @Cached("createIsInstantiableNode()") Node isInstantiableNode) {
+            return ForeignAccess.sendIsInstantiable(isInstantiableNode, receiver);
+        }
+
+        protected Node createIsInstantiableNode() {
+            return Message.IS_INSTANTIABLE.createNode();
+        }
+
+        @Fallback
+        public boolean isInstantiable(Object receiver) {
+            return false;
+        }
```

Comment:
Style: usually specializations are kept together and the helper methods are below, so it's clear which specializations are in the node.

Reply:
Not relevant any more.